File: internal/cmd/paths.go
package cmd

import (
	"fmt"
	"os"
	"path/filepath"
)

// getBaseDir returns the current working directory or an error
func getBaseDir() (string, error) {
	cwd, err := os.Getwd()
	if err != nil {
		return "", fmt.Errorf("failed to get working directory: %w", err)
	}
	return cwd, nil
}

// getFeedsDir returns the feeds directory path
func getFeedsDir() (string, error) {
	cwd, err := getBaseDir()
	if err != nil {
		return "", err
	}
	return filepath.Join(cwd, "exchanges", "feeds"), nil
}

// getSchemasDir returns the schemas directory path
func getSchemasDir() (string, error) {
	cwd, err := getBaseDir()
	if err != nil {
		return "", err
	}
	return filepath.Join(cwd, "exchanges", "schemas"), nil
}

// getEnvsDir returns the environments directory path
func getEnvsDir() (string, error) {
	cwd, err := getBaseDir()
	if err != nil {
		return "", err
	}
	return filepath.Join(cwd, "exchanges", "environments"), nil
}

File: ssmd-agent/src/cli/utils/paths.ts
// Path utilities for finding ssmd directories
import { join } from "https://deno.land/std@0.224.0/path/mod.ts";

/**
 * Find the exchanges root directory by walking up from cwd
 */
export async function findExchangesRoot(): Promise<string> {
  let dir = Deno.cwd();

  while (dir !== "/" && dir !== "") {
    try {
      const exchangesPath = join(dir, "exchanges");
      const stat = await Deno.stat(exchangesPath);
      if (stat.isDirectory) {
        return exchangesPath;
      }
    } catch {
      // Not found, keep looking
    }

    const parent = join(dir, "..");
    if (parent === dir) break;
    dir = parent;
  }

  throw new Error("exchanges directory not found. Run 'ssmd init' first.");
}

/**
 * Get the feeds directory path
 */
export async function getFeedsDir(): Promise<string> {
  const root = await findExchangesRoot();
  return join(root, "feeds");
}

/**
 * Get the schemas directory path
 */
export async function getSchemasDir(): Promise<string> {
  const root = await findExchangesRoot();
  return join(root, "schemas");
}

/**
 * Get the environments directory path
 */
export async function getEnvsDir(): Promise<string> {
  const root = await findExchangesRoot();
  return join(root, "environments");
}
